homeassistant/components/opentherm_gw/entity.py

```diff
@@ -0,0 +1,67 @@
+"""Common opentherm_gw entity properties."""
+
+from dataclasses import dataclass
+import logging
+
+from homeassistant.core import callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, TRANSLATE_SOURCE
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True)
+class OpenThermEntityDescriptionMixin:
+    """Mixin for common opentherm_gw entity properties."""
+
+    friendly_name_format: str
+
+
+class OpenThermEntity(Entity):
+    """Represent an OpenTherm Gateway entity."""
+
+    _attr_should_poll = False
+    _attr_entity_registry_enabled_default = False
+    _attr_available = False
+
+    def __init__(self, gw_dev, source, description):
+        """Initialize the entity."""
+        self.entity_description = description
+        self._gateway = gw_dev
+        self._source = source
+        friendly_name_format = (
+            f"{description.friendly_name_format} ({TRANSLATE_SOURCE[source]})"
+            if TRANSLATE_SOURCE[source] is not None
+            else description.friendly_name_format
+        )
+        self._attr_name = friendly_name_format.format(gw_dev.name)
```

Comment:
Let's check if we can move to translated names in follow ups :)

Reply:
I have some follow-ups planned, this will be added to the list.